### PR TITLE
business logic for quantity update and deletion of product

### PR DESCRIPTION
**Pull request by PremiyaR**

### CODE REVIEW

File: inventory/src/main/java/com/pilot/inventory/controller/UsersController.java [hunk 1]
```diff
@@ -1,10 +1,45 @@
 package com.pilot.inventory.controller;
 
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.pilot.inventory.model.entity.Users;
+import com.pilot.inventory.service.UsersService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
```
Comment (badCODERever) [inline inventory/src/main/java/com/pilot/inventory/controller/UsersController.java:9]: remove wildcard operator and import only required classes

File: inventory/src/main/java/com/pilot/inventory/controller/UsersController.java [hunk 2]
```diff
@@ -1,10 +1,45 @@
...
+
+import java.util.List;
 
 @RestController
 @RequestMapping("/users")
```
Comment (badCODERever) [inline inventory/src/main/java/com/pilot/inventory/controller/UsersController.java:14]: Better to use constants and (can create new constant class to maintain all the REST constants)

File: inventory/src/main/java/com/pilot/inventory/model/entity/Categories.java [hunk 3]
```diff
@@ -2,15 +2,14 @@
 
 import jakarta.persistence.*;
```
Comment (badCODERever) [inline inventory/src/main/java/com/pilot/inventory/model/entity/Categories.java:3]: remove wildcard operator use only required class

File: inventory/src/main/java/com/pilot/inventory/model/entity/Product.java [hunk 4]
```diff
@@ -17,14 +17,12 @@ public class Product {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private int id;
 
-
     private String name;
 
     @ManyToOne
-    @JoinColumn(nullable = false,name = "category_id")
```
Comment (badCODERever) [inline inventory/src/main/java/com/pilot/inventory/model/entity/Product.java:24]: why is one is removed?

File: inventory/src/main/java/com/pilot/inventory/service/OrdersServiceImpl.java [hunk 5]
```diff
@@ -22,40 +25,47 @@ public class OrdersServiceImpl implements OrdersService{
...
         {
-            ordersRepository.deleteById(id);
-            status="Product deleted";
+            Orders orders = optional.get();
+            orders.setDeleted(true);
+            ordersRepository.save(orders);
+            return "Order deleted";
         }
         else {
-            status="Product not deleted,Please check your id is valid";
```
Comment (badCODERever) [inline inventory/src/main/java/com/pilot/inventory/service/OrdersServiceImpl.java:48]: Could you please explain, this block of code required?

**Review by badCODERever — CHANGES_REQUESTED** [review]

File: inventory/src/main/java/com/pilot/inventory/service/OrdersServiceImpl.java (hunk 5 above)
Comment (PremiyaR) [reply inventory/src/main/java/com/pilot/inventory/service/OrdersServiceImpl.java:48]: If the given ID doesn't exist, the row cannot be found and deleted. Hence why I wrote this code.

File: inventory/src/main/java/com/pilot/inventory/model/entity/Product.java (hunk 4 above)
Comment (PremiyaR) [reply inventory/src/main/java/com/pilot/inventory/model/entity/Product.java:24]: I encountered an error in between, so I tried removing it. Forgot to add it back.

**Pull request merged**